<a href="https://colab.research.google.com/github/theritik08/Ritik-Flyrank/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/theritik08/Ritik-Flyrank/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Going with Lane 4 — CTR/Engagement Opportunity Scoring.

Honestly this is the one that clicked for me because of something I already
noticed in notebook 01. When I looked at CTR by position tier, it basically
falls off a cliff — pages at the top get ~35% CTR, pages buried deep barely
get 5-6%. So if I just sorted all pages by raw CTR, I'd mostly just be
re-discovering "position matters," not finding anything useful.

What's actually interesting is comparing a page to *other pages at the same
position* — is this specific page underperforming for where it sits, or is
it just doing what's expected for that position? That felt like a real
question worth digging into, not just a metric to report.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Unit of analysis is a single page.

The decision: which pages a reviewer should actually spend time on first,
out of way more pages than anyone has time to check manually. Right now
that's probably just gut feeling or whoever complains loudest.

Who acts on it: whoever's doing content/SEO review — they'd get a shortlist
instead of scrolling a spreadsheet.

The action: reviewer looks at the page, maybe rewrites the title or meta
description, or checks if it's even still answering what people are
searching for.

Cost if I get it wrong: if I flag a page that's actually fine, that's
wasted review time — and worse, it means the real problem page didn't get
looked at instead. If I miss an actual underperformer, nobody notices and
the client just keeps losing clicks they could've had.

Why this needs data instead of just eyeballing it: because raw CTR is
mostly explained by position, a person skimming a list of low-CTR pages
would mostly just be finding low-position pages, not actual problems.
Adjusting for position first is the part a person wouldn't naturally do
by hand across thousands of rows.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd, numpy as np

df = pd.read_csv("https://raw.githubusercontent.com/theritik08/Ritik-Flyrank/refs/heads/main/data/raw/content_refresh_anonymized.csv")
print(df.shape[0], "rows,", df.shape[1], "columns")

# Number 1: CTR by position tier
visible = df[df["impressions_90d"] >= 100]
tier_ctr = visible.groupby("position_tier")["ctr"].mean().sort_values(ascending=False)
print("\nMean CTR by position tier:")
print(tier_ctr.round(4).to_string())

# Number 2: pages noticeably below their own tier's average CTR
visible500 = df[df["impressions_90d"] >= 500].copy()
tier_mean = visible500.groupby("position_tier")["ctr"].transform("mean")
visible500["ctr_gap_pct"] = (visible500["ctr"] - tier_mean) / tier_mean

underperformers = visible500[visible500["ctr_gap_pct"] <= -0.5]
print(f"\nVisible pages (impressions_90d >= 500): {len(visible500):,}")
print(f"Pages at least 50% below their tier's average CTR: {len(underperformers):,} "
      f"({len(underperformers)/len(visible500):.1%})")

print(f"Total impressions in underperforming pages: {underperformers['impressions_90d'].sum():,.0f}")

30000 rows, 44 columns

Mean CTR by position tier:
position_tier
page_1      0.3548
top_3       0.3341
striking    0.2558
page_3_5    0.1424
deep        0.0554

Visible pages (impressions_90d >= 500): 16,726
Pages at least 50% below their tier's average CTR: 6,727 (40.2%)
Total impressions in underperforming pages: 49,827,273


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can say:some pages have noticeably lower CTR than others at the
same position — that's just a measured pattern. About 40% of visible
pages (16,726) sit 50%+ below their tier's average CTR, with ~50M
impressions tied up in them.

One caveat: 40% feels high. I compared to the tier *mean*, and CTR is
probably skewed — a few high-CTR pages could be pulling that average up.
I'd want to re-check against the median before trusting that number fully.
This is just the exploratory pass showing the lane has signal, not a
finished score.

What I can't say: that fixing a flagged page will actually help — no
before/after data, no experiment. I also don't know *why* a page
underperforms without looking closer. And I'm not claiming anything about
how Google's algorithm works — this is pattern-spotting in FlyRank's data,
nothing more.

No real titles, URLs, or client names in any output — that data isn't in
this dataset to begin with, it's all pseudonymized.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.